src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -2215,219 +2233,228 @@ void PasswordSafeFrame::OnUpdateUI(wxUpdateUIEvent& evt)
-      evt.Enable(!isFileReadOnly && m_core.IsDbFileSet());
+      evt.Enable(isUnlocked && !isFileReadOnly && m_core.IsDbFileSet());
       break;
 
     case wxID_FIND:
-      evt.Enable(m_core.IsDbFileSet() && m_core.GetNumEntries() != 0);
+      evt.Enable(isUnlocked && m_core.IsDbFileSet() && m_core.GetNumEntries() != 0);
       break;
 
     case wxID_ADD:
-      evt.Enable(!isFileReadOnly && m_core.IsDbFileSet());
+      evt.Enable(isUnlocked && !isFileReadOnly && m_core.IsDbFileSet());
       break;
 
     case wxID_DELETE:
-      evt.Enable(!isFileReadOnly && ((pci && !pci->IsProtected()) || isTreeViewGroupSelected));
+      evt.Enable(isUnlocked && !isFileReadOnly && ((pci && !pci->IsProtected()) || isTreeViewGroupSelected));
       break;
 
     case ID_DUPLICATEENTRY:
-      evt.Enable(!isFileReadOnly && pci);
+      evt.Enable(isUnlocked && !isFileReadOnly && pci);
       break;
 
     case ID_SHOWHIDE_UNSAVED:
-      evt.Enable((m_CurrentPredefinedFilter == UNSAVED) || ((m_CurrentPredefinedFilter == NONE) && m_core.IsDbFileSet() && !isFileReadOnly && m_core.HasDBChanged()));
+      evt.Enable(isUnlocked && ((m_CurrentPredefinedFilter == UNSAVED) || ((m_CurrentPredefinedFilter == NONE) && m_core.IsDbFileSet() && !isFileReadOnly && m_core.HasDBChanged())));
       evt.Check(m_CurrentPredefinedFilter == UNSAVED);
       break;
 
     case ID_SHOW_ALL_EXPIRY:
-      evt.Enable((m_CurrentPredefinedFilter == EXPIRY) || ((m_CurrentPredefinedFilter == NONE) &&
+      evt.Enable(isUnlocked && ((m_CurrentPredefinedFilter == EXPIRY) || ((m_CurrentPredefinedFilter == NONE) &&
        m_core.IsDbFileSet() &&
-       m_core.GetExpirySize() != 0));
+       m_core.GetExpirySize() != 0)));
       evt.Check(m_CurrentPredefinedFilter == EXPIRY);
       break;
 
     case ID_SHOW_LAST_FIND_RESULTS:
-      evt.Enable((m_CurrentPredefinedFilter == LASTFIND) || ((m_CurrentPredefinedFilter == NONE) &&
+      evt.Enable(isUnlocked && ((m_CurrentPredefinedFilter == LASTFIND) || ((m_CurrentPredefinedFilter == NONE) &&
                   m_core.IsDbFileSet() &&
-                  m_FilterManager.GetFindFilterSize() != 0));
+                  m_FilterManager.GetFindFilterSize() != 0)));
       evt.Check(m_CurrentPredefinedFilter == LASTFIND);
       break;
 
     case ID_MERGE:
     case ID_IMPORTMENU:
-      evt.Enable(!isFileReadOnly && m_core.IsDbFileSet());
+      evt.Enable(isUnlocked && !isFileReadOnly && m_core.IsDbFileSet());
       break;
-      
+
     case ID_IMPORT_XML:
 #if (!defined(_WIN32) && USE_XML_LIBRARY == MSXML)
       evt.Enable(false);
 #else
-      evt.Enable(!isFileReadOnly && m_core.IsDbFileSet());
+      evt.Enable(isUnlocked && !isFileReadOnly && m_core.IsDbFileSet());
 #endif
       break;
 
     case ID_PROTECT:
-      evt.Enable(!isFileReadOnly && pci && !pci->IsShortcut());
+      evt.Enable(isUnlocked && !isFileReadOnly && pci && !pci->IsShortcut());
       evt.Check(pci && pci->IsProtected());
       break;
```

Comment:
We can also use the `isUnlocked` variable below on lines 2468 and 2472 instead of `!m_sysTray->IsLocked()`.

Reply:
I have changed this as suggested.